server/pulp/server/controllers/repository.py

```diff
@@ -165,16 +166,23 @@ def find_units_not_downloaded(repository):
     :return: The requested units.
     :rtype: generator
     """
-    types = {}
-    q_set = model.RepositoryContentUnit.objects(repo_id=repository.repo_id)
-    for association in q_set.only('unit_id', 'unit_type_id'):
-        unit_ids = types.setdefault(association.unit_type_id, [])
-        unit_ids.append(association.unit_id)
-    for type_id, unit_ids in types.items():
-        _model = plugin_api.get_unit_model_by_id(type_id)
-        for page in paginate(unit_ids):
-            for unit in _model.objects(id__in=page, downloaded=False):
-                yield unit
+    query_sets = _get_mongoengine_repo_querysets(str(repository.repo_id))
+    query_sets = [q(downloaded=False) for q in query_sets]
+    return chain(*query_sets)
+
+
+def missing_unit_count(repository):
```

Comment:
This is up for debate, but my feeling is that if there is zero chance that this function is going to need the full repository object, then just let it take the ``repo_id`` as its argument. It would be a shame to make someone retrieve the full object from the DB just for this function, which would ignore everything but the id.